test/integration/scheduler/scheduler_test.go

```diff
@@ -174,22 +174,96 @@ func TestSchedulerCreationFromNonExistentConfigMap(t *testing.T) {
 	ss := options.NewSchedulerServer()
 	ss.PolicyConfigMapName = "non-existent-config"
 
-	_, err := app.CreateScheduler(ss, clientSet,
-		informerFactory.Core().V1().Nodes(),
-		informerFactory.Core().V1().PersistentVolumes(),
-		informerFactory.Core().V1().PersistentVolumeClaims(),
-		informerFactory.Core().V1().ReplicationControllers(),
-		informerFactory.Extensions().V1beta1().ReplicaSets(),
-		informerFactory.Apps().V1beta1().StatefulSets(),
-		informerFactory.Core().V1().Services(),
-		eventBroadcaster.NewRecorder(api.Scheme, clientv1.EventSource{Component: v1.DefaultSchedulerName}),
-	)
+	_, err := app.CreateScheduler(ss, clientSet, informerFactory,	eventBroadcaster.NewRecorder(api.Scheme, clientv1.EventSource{Component: v1.DefaultSchedulerName}))
 
 	if err == nil {
 		t.Fatalf("Creation of scheduler didn't fail while the policy ConfigMap didn't exist.")
 	}
 }
 
+func TestChangeSchedulerConfigMap(t *testing.T) {
+	_, s := framework.RunAMaster(nil)
+	defer s.Close()
+
+	ns := framework.CreateTestingNamespace("configmap", s, t)
+	defer framework.DeleteTestingNamespace(ns, s, t)
+
+	clientSet := clientset.NewForConfigOrDie(&restclient.Config{Host: s.URL, ContentConfig: restclient.ContentConfig{GroupVersion: &api.Registry.GroupOrDie(v1.GroupName).GroupVersion}})
+	defer clientSet.Core().Nodes().DeleteCollection(nil, metav1.ListOptions{})
+	informerFactory := informers.NewSharedInformerFactory(clientSet, 0)
+
+	sched, err := CreateSchedulerFromConfigMap(clientSet, informerFactory)
+	if err != nil {
+		t.Fatalf("Error creating scheduler: %v", err)
+	}
+
+	var schedulerGotKilledAfterConfigMapUpdate bool = false
+	app.SchedulerKillFunc = func () {
+		t.Log("Scheduler killer function got called.")
+		schedulerGotKilledAfterConfigMapUpdate = true
+	}
+
+	informerFactory.Start(sched.Config().StopEverything)
+	defer close(sched.Config().StopEverything)
+
+	sched.Run()
+
+	newPolicyConfigMap := v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{Namespace: metav1.NamespaceSystem, Name: defaultSchedulerPolicyConfigMap},
+		Data: map[string]string{
+			options.SchedulerPolicyConfigMapKey: `{
+			"kind" : "Policy",
+			"apiVersion" : "v1",
+			"predicates" : [
+				{"name" : "PredicateOne"}
+			],
+			"priorities" : [
+				{"name" : "PriorityOne", "weight" : 2},
+				{"name" : "PriorityTwo", "weight" : 3}
+			]
+			}`,
+		},
+	}
+	// Update the ConfigMap.
+	_, err = clientSet.Core().ConfigMaps(metav1.NamespaceSystem).Update(&newPolicyConfigMap)
+	if err != nil {
+		t.Fatalf("Error updating scheduler policy configmap: %v", err)
+	}
+	time.Sleep(2 * time.Second)
+	if !schedulerGotKilledAfterConfigMapUpdate {
+		t.Error("Scheduler killer function is expected to be called upon update to its policy ConfigMap.")
+	}
+
+	// Scheduler should be killed when its policy ConfigMap is deleted.
```

Comment:
The test tests both update and restart. Well, the update part needs a comment, but I can split the two.